packages/manager/modules/pci/src/projects/project/kubernetes/components/autoscaling/autoscaling.controller.js

```diff
@@ -0,0 +1,195 @@
+import { AUTOSCALING_LINK } from './autoscaling.constants';
+import {
+  SCALE_DEFAULT_VALUES,
+  ANTI_AFFINITY_MAX_NODES,
+} from '../../kubernetes.constants';
+
+export default class AutoscalingController {
+  /* @ngInject */
+  constructor($scope, coreConfig) {
+    this.$scope = $scope;
+    this.ovhSubsidiary = coreConfig.getUser().ovhSubsidiary;
+    this.autoscalingInfoLink = this.getAutoscalingInfoLink();
+    this.LOWEST_VALUE = SCALE_DEFAULT_VALUES.LOWEST_VALUE;
+  }
+
+  $onInit() {
+    this.model = this.nodePool.autoscaling;
+    const { lowest, desired, highest } = this.model.nodes;
+
+    this.init(true);
+    this.setAutoscalingValidity(lowest.value, desired.value, highest.value);
+    this.$scope.$watch('$ctrl.nodePool.antiAffinity', () =>
+      this.onAntiAffinityChange(),
+    );
```

Comment:
Can't we use something else ? Like a `onChange` binding ?

Reply:
No Marie, the `antiAffinity` is implemented into another step component.
I need it here to follow the changes when customer change the `antiAffinity` in another step.

Reply:
Is it inside a `oui-step-form` component, in that case maybe we can check when we enter the step that needs the value no ?

Reply:
Yes, is't inside a `oui-step-form`.
No i can't, because the autoscaling is a few step before the antiAffinity step.

If customer set the antiaffinity we must set the autosacling limits.
I hope it's more clear @marie-j.

Reply:
I'm having trouble understanding why using a `onChange` binding that updates `$ctrl.model.autoscaling` is not possible even if the step is before (also I'm not sure about the experience here, shouldn't antiaffinity and autoscaling be in another order or in the same step ?

Reply:
Because the switch component is inside a dedicated step component and the autoscaling has it dedicated component.
The both are not connected one to each other. To be able to be follow the changes i used $watch.

About the step, Maxime said it should stay as we already QC it.

Reply:
Simplified example of what I had in mind: 
```
//template.html
<oui-step-form>
 <anti-affinity-component model="$ctrl.antiAffinityModel"></anti-affinity-component>
</oui-step-form>
<oui-step-form>
 <component
  on-change="$ctrl.onChange()>
 </component>
</oui-step-form>

//controller.js
onChange() {
 this.antiAffinityModel= someValue 
}

//antiaffinity.component.js
export default {
 bindings: {
  antiAffinityModel: '<'
 }
}
```

Reply:
I have read more about $watch, and it's a not an angularjs antipattern.
I think we can let it @marie-j.